app/presenters/blacklight/document_presenter.rb

```diff
@@ -11,10 +11,12 @@ class DocumentPresenter
     # @param [SolrDocument] document
     # @param [ActionView::Base] view_context scope for linking and generating urls
     # @param [Blacklight::Configuration] configuration
-    def initialize(document, view_context, configuration = view_context.blacklight_config)
+    # @param [Integer] counter what offset in the search result is this record (used for tracking)
+    def initialize(document, view_context, configuration = view_context.blacklight_config, counter: nil)
```

Comment:
I guess this is ok for backwards-compatibility? Maybe it'd be better to check arity in the helpers? Or just call it out in the release notes, because there are at least a couple projects on github that overrode `initialize`..

Reply:
Because this is an optional argument, the arity hasn't changed.

Reply:
Yes, that seems to work fine.

Reply:
Now that we're developing for 8.0 on main: Should `configuration` also be a keyword param? Something about the mix of optional positional args and optional keyword args seems fishy to me.

Reply:
@barmintor I think that is fine, but does that belong in this PR or should that be a separate change?

Reply:
that's a good point